### DISCUSSION TITLE
Interaction of bundle & "critical" attribute with OSCQuery

### DISCUSSION THREAD

**Discussion by jcelerier** [OSSIA/libossia#712] [Q&A]
Hello,
let's say I have the following tree : 

/volume (float)
/play (bool, critical)

What happens if the user wants to create an OSC bundle with both ? The most logical thing to me would be to send everything in a single bundle over TCP, would that be fine ?   (the alternative being, splitting things over TCP and UDP depending on critical-ness, but then you loose any packing that you may have wanted to keep)

**Comment by jln-** [comment]
> The most logical thing to me would be to send everything in a single bundle over TCP

Agree with that.

Pinging @matcham

**Reply by matcham** [reply]
osc over TCP ?
I know that in theory it's possible.. but in practice who actually uses that ?
I don't know any osc-enabled software that can be configured to send/receive osc over tcp.
even in Max, there's no native tcpreceive / tcpsend object (can be done using java, but :-(
and if some messages are sent over udp and other over tcp, how are we supposed to set up ports for each protocol ?
so I would say : no tcp for osc because I don't know how to receive them !

**Reply by maybites** [reply]
there are these libraries for max: http://www.sadam.hu/en/software

and in theory you could use a combination of [sadam.tcpReceiver] -> [sadam.udpSender] -> [udpreceive] to get the osc messages.

**Reply by jcelerier** [reply]
@matcham 

> I know that in theory it's possible.. but in practice who actually uses that ?

everyone who uses libossia with oscquery and the "critical" attribute set on parameters :-)

**Reply by matcham** [reply]
@jcelerier  : and what about those who use osc(non-query) with other softwares that support only osc over udp ?

**Reply by jcelerier** [reply]
this topic is specifically about oscquery, which does use both TCP and UDP. If you use the osc protocol only in libossia (e.g. with `[expose osc ...>` then everything goes through UDP. And if you use OSCQuery then you likely use it on both sides, you're not using any udpsend / ... anywhere.

**Reply by jcelerier** [reply]
(updated the title to make that more clear)

**Reply by matcham** [reply]
ho, ok, so i don't care :-)
